classes/api.php

```diff
@@ -356,18 +361,83 @@ private static function validate_quiz(?object $quiz): array {
         }
 
         // Are any of the questions a disallowed qtype ?
-        $invalidqtypequestions = array_filter($questions, function($q) {
-            return !in_array($q->qtype, self::SUPPORTED_QTYPES);
-        });
+        $invalidqtypequestions = self::contains_invalid_qtype($questions);
 
-        if (!empty($invalidqtypequestions)) {
+        if ($invalidqtypequestions === true) {
```

Comment:
Should the following work:
```suggestion
        if (self::contains_invalid_qtype($questions)) {
```

Reply:
Ah yep it used to return an array but now returns a bool, so this works and is cleaner :+1: